capellambse/model/common/accessors.py

```diff
@@ -1201,16 +1272,23 @@ def __setitem__(
         self, index: int | slice, value: T | cabc.Iterable[T]
     ) -> None:
         assert self._parent is not None
-        del self[index]
-        if isinstance(index, slice):
-            assert isinstance(value, cabc.Iterable)
-            for i, elm in enumerate(value, start=index.start or 0):
-                self.insert(i, elm)
-        else:
-            assert not isinstance(value, cabc.Iterable)
-            self.insert(index, value)
+        accessor = type(self)._accessor
+        assert isinstance(accessor, WritableAccessor)
+
+        new_objs = list(self)
+        new_objs[index] = value  # type: ignore[index]
+
+        if self.fixed_length and len(new_objs) != self.fixed_length:
+            raise TypeError(
+                f"Cannot set: List must stay at length {self.fixed_length}"
+            )
```

Comment:
Isn't a `ValueError` more suited for this? `value`s length doesn't match to this `ElementList.fixed_length`.

Reply:
The idea behind TypeError is that "this type of object doesn't allow this operation" (which I've seen used in this way in a few places). And it makes sense because it's a different type of "CoupledElementList" than one that would allow it (because `fixed_length` is a `ClassVar`). But "the value you provided doesn't fit" works just as well ... I don't know, I'm biased towards mine, so you call it. :P

Reply:
Your explanation fits.